### DISCUSSION TITLE
[core] about inter-procedural analysis

### DISCUSSION THREAD

**Discussion by tiandiyixian** [pmd/pmd#3047] [Q&A] [answered]
**Affects PMD Version:** 
6.0
**Rule:**

**Description:**
Does pmd dfa support inter-procedural  analysis?such as cross function  or cross file.I want to realize a taint analysis about tracing the data flow  between source and  sink .
thanks
**Code Sample demonstrating the issue:**

```

```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by jsotuyod** [comment, marked as answer]
@tiandiyixian PMD currently doesn't perform inter procedural analysis. It's in [our roadmap](https://github.com/pmd/pmd/wiki/Project-Ideas-%5BInception%5D#data-flow-analysis-and-control-flow-graph) and we are even considering proposing some work on DFA as part of Google Summer of Code 2018.

As always, all PRs and contributions to help move this forward faster are greatly welcomed.

**Reply by ZH3FENG** [reply]
Is there any change?
It seems still does not supported yet?

**Reply by jsotuyod** [reply]
no changes yet, contributions welcomed!
